scripts/circleci_scraper/scraper.py

```diff
@@ -270,51 +274,70 @@ def export_test_artifacts_by_job(
             )
             if not artifacts:
                 break
-            test_artifacts += [item for item in artifacts.items if item.path.endswith(".xml")]
+            for item in artifacts.items:
+                filename = Path(item.path).name
+                if filename.endswith(".xml"):
+                    junit_artifacts.append(item)
+                elif re.match(COVERAGE_FILE_REGEX, filename, re.IGNORECASE):
+                    coverage_artifacts.append(item)
             next_page_token = artifacts.next_page_token
             if not next_page_token:
                 break
-        self.export_test_artifacts(repository, workflow_name, job, test_artifacts)
+        if junit_artifacts:
+            self.export_artifacts(
+                repository, workflow_name, self._junit_artifact_dir, job, junit_artifacts
+            )
+        if coverage_artifacts:
+            self.export_artifacts(
+                repository, workflow_name, self._coverage_artifact_dir, job, coverage_artifacts
+            )
 
-    def export_test_artifacts(
-        self, repository: str, workflow_name: str, job: Job, artifacts: list[Artifact]
+    def export_artifacts(
+        self,
+        repository: str,
+        workflow_name: str,
+        artifact_directory: str,
+        job: Job,
+        artifacts: list[Artifact],
     ):
         """Export a given list of test artifacts.
 
         Args:
             repository (str): The repository name.
             workflow_name (str): The workflow name.
+            artifact_directory (str): The destination directory name for artifacts.
             job (Job): The job details.
             artifacts (list[Artifact]): The list of artifacts.
 
         Raises:
             CircleCIClientError: If there is an error in the CircleCI API request.
             CircleCIScraperError: If there is an error in downloading the artifacts.
         """
-        artifact_directory = (
+        artifact_path = (
             Path(self._test_result_dir)
             / repository
             / workflow_name
             / job.name
-            / self._test_artifact_dir
+            / artifact_directory
             / str(job.job_number)
         )
-        if artifacts:
-            artifact_directory.mkdir(parents=True, exist_ok=True)
+        if artifact_path.exists():
+            self.logger.info(f"{artifact_path} already exists, skipping download(s).")
+            return
+
+        artifact_path.mkdir(parents=True)
         for index, artifact in enumerate(artifacts):
-            file_path = artifact_directory / f"{index}-{Path(artifact.path).name}"
-            if file_path.exists():
-                self.logger.info(f"{file_path} already exists, skipping download.")
-            else:
-                self.download_artifact(str(file_path), artifact.url)
+            file_path = artifact_path / f"{index}-{Path(artifact.path).name}"
+            self.download_artifact(str(file_path), artifact.url)
```

Comment:
Does this need to be `str` or could we use an f-string?

Reply:
🤔 I think we can just pass the Path come to thing of it... updated